src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/GenericCompositeBindingProvider.cs

```diff
@@ -67,5 +69,34 @@ public GenericCompositeBindingProvider(IEnumerable<IBindingProvider> providers)
             }
             throw new InvalidOperationException("Can't bind " + resourceName + " to type '" + context.Parameter.ParameterType + "'.");
         }
+
+        public IEnumerable<Rule> GetRules()
+        {
+            foreach (var provider in _providers.OfType<IRuleProvider>())
```

Comment:
nit: could use the following, but it probably doesn't improve readability (I like linq too much)

```
_providers
  .OfType<IRuleProvider>()
  .SelectMany(rule => rule.GetRules())
```